src/classes/user/UserRepository.js

```diff
@@ -1,23 +1,46 @@
-import axios from 'axios'
+import axios from "axios"
+import UserMapper from "./UserMapper"
 
 export default class UserRepository {
+
 	async getUsers() {
-		const users = (await axios.get('http://localhost:3000/users')).data
-		return users
+		const users = (await axios.get('http://localhost:3000/users')).data;
+		return users.map(UserMapper.map)
 	}
 
-	async getUserNickname(id) {
-		const nickname = (await axios.get('http://localhost:3000/users?id=' + id)).data[0].nick_name;
-		return nickname
+	async signIn(credential) {
+		const postResponse = (await axios.post('http://localhost:3000/auth/login', credential))
+		return postResponse
 	}
 
-	createUser(credential) {
-		axios.post('http://localhost:3000/users', {credential})
+	async getUserById(id) {
+		const user = (await axios.get('http://localhost:3000/users/' + id, {
+			headers: {
+				authorization: localStorage.getItem('token')
+			}
+		})).data;
+		return UserMapper.map(user);
 	}
 
-	async signIn(credential) {
-		const token = (await axios.post('http://localhost:3000/auth/login', {credential})).data.access_token
-		return token
+	async findUsers(nick_name) {
+		const user = (await axios.get('http://localhost:3000/users?nick_name_like=' + nick_name)).data;
+		return user.map(UserMapper.map);
+	}
+
+	createUser(user) {
```

Comment:
Not async again, what if server down or some logic broken?

Reply:
add 'async', add new task for checking  response